lib/kaffe/group_member/manager/group_manager.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Kaffe.GroupManager do
+  @moduledoc """
+  This is the main process for launching group members and workers.
+
+  The process begins by starting the client connection to Kafka. Then, group
+  members are created for each of the configured topics.
+
+  This is also the place where subscribers get the PID for a Worker.
+  """
+  
+  use GenServer
+
+  alias Kaffe.GroupMemberSupervisor
+  alias Kaffe.WorkerSupervisor
+
+  require Logger
+
+  defmodule State do
+    defstruct supervisor_pid: nil,
+      subscriber_name: nil,
+      consumer_group: nil,
+      topics: nil,
+      offset: nil
+  end
+
+  def start_link() do
+    GenServer.start_link(__MODULE__, [self()])  
+  end
+
+  ## Callbacks
+
+  def init([supervisor_pid]) do
+    Logger.info "event#startup=#{inspect self()}"
+
+    config = Kaffe.Config.Consumer.configuration
+
+    :ok = kafka().start_client(config.endpoints, config.subscriber_name, config.consumer_config)
+
+    send(self(), {:start_group_members})
+
+    {:ok, %State{supervisor_pid: supervisor_pid,
+                subscriber_name: config.subscriber_name,
+                consumer_group: config.consumer_group,
+                topics: config.topics,
+                offset: Kaffe.Config.Consumer.begin_offset}}
+  end
+
+  def handle_info({:start_group_members}, state) do
```

Comment:
Why is this coming in as a process message instead of a typical GenServer call or cast? Skipping call/cast can be necessary, but there are tradeoff like losing timeouts and backpressure.